src/components/ListMessages.tsx

```diff
@@ -0,0 +1,50 @@
+import { MessageResponse } from "@frequency-chain/api-augment/interfaces";
+import { Button, Space, Table } from "antd";
+import Column from "antd/lib/table/Column";
+import React from "react";
+import { fetchMessagesForSchema } from "../services/chain/apis/extrinsic";
+import { MessageDetails, SchemaProps } from "../services/types";
+import { staticSchema } from "./RegisterSchema";
+
+const ListMessages = (props: SchemaProps): JSX.Element => {
+
+    const [listOfMessages, setListOfMessage] = React.useState<MessageDetails[]>([]);
+
+    const listMessages = async () => {
+        const messages: MessageResponse[] = await fetchMessagesForSchema(parseInt(props.schema.schema_id));
+
+        let allMessages: MessageDetails[] = messages.map((msg) => {
+            return {
+                key: Number(msg.index) + 1,
```

Comment:
The index on msg is the index inside the block number. So combining them like: msg.index + "_" + msg.block_number would be unique, or you can just use the index from the map:

```suggestion
        let allMessages: MessageDetails[] = messages.map((msg, index) => {
            return {
                key: index,
```